src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -274,6 +275,40 @@ internal BinaryClassificationTrainers(BinaryClassificationCatalog catalog)
                 Evaluate(x.Scores, labelColumn), x.Scores, x.Fold)).ToArray();
         }
 
+        /// <summary>
+        /// Change threshold for binary model.
+        /// </summary>
+        /// <typeparam name="TModel">An implementation of the <see cref="IPredictorProducing{TResult}"/></typeparam>
+        /// <param name="chain">Chain of transformers.</param>
+        /// <param name="threshold">New threshold.</param>
+        /// <returns></returns>
+        public TransformerChain<BinaryPredictionTransformer<TModel>> ChangeModelThreshold<TModel>(TransformerChain<BinaryPredictionTransformer<TModel>> chain, float threshold)
+            where TModel : class
+        {
+            if (chain.LastTransformer.Threshold == threshold)
+                return chain;
+            List<ITransformer> transformers = new List<ITransformer>();
+            var predictionTransformer = chain.LastTransformer;
+            foreach (var transform in chain)
+            {
+                if (transform != predictionTransformer)
+                    transformers.Add(transform);
+            }
+
+            transformers.Add(new BinaryPredictionTransformer<TModel>(Environment, predictionTransformer.Model,
+                predictionTransformer.TrainSchema, predictionTransformer.FeatureColumn,
+                threshold, predictionTransformer.ThresholdColumn));
+            return new TransformerChain<BinaryPredictionTransformer<TModel>>(transformers.ToArray());
+        }
+
+        public BinaryPredictionTransformer<TModel> ChangeModelThreshold<TModel>(BinaryPredictionTransformer<TModel> model, float threshold)
+             where TModel : class
+        {
+            if (model.Threshold == threshold)
```

Comment:
>if (model.Threshold == threshold) [](start = 12, length = 33)

do you want to warn here?

Reply:
I think we should provide the same warning that C# does when you have a variable like `int a = 5` and then assign `5` to it later.

---
In reply to: [265862991](https://github.com/dotnet/machinelearning/pull/2969#discussion_r265862991) [](ancestors = 265862991)